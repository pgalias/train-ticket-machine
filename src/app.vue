<template>
  <station-chooser />
</template>

<script>
import LocalStorage from './infrastructure/cache/local-storage';
import StationChooser from './app/views/station-chooser/station-chooser.vue';

/**
 * TODO:
 *  - Add dependency injection library for services
 *  - Add router for view(s)
 *  - POST MVP: Add translations
 *  - POST MVP: Add virtual list for stations list
 */

export default {
  name: 'App',
  components: {
    StationChooser,
  },
  provide() {
    return {
      cache: new LocalStorage(),
    };
  },
};
</script>
