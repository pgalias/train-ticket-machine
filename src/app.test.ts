import { shallowMount } from '@vue/test-utils';
import App from './app.vue';

describe('App', () => {
  test('should match snapshot', () => {
    // @ts-ignore
    const wrapper = shallowMount(App);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
