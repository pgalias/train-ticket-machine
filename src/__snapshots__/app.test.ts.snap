// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `<station-chooser-stub></station-chooser-stub>`;
